### ISSUE TITLE
gui enhancement - mute groups visualisation

### ISSUE THREAD

**Issue by a13** [hydrogen-music/hydrogen#53]
I think It would be useful to have an option to group instruments in the instruments/pattern editor by mute group. The group should be able to be added( _instruments_ submenu)/deleted/collapsed(-)/expanded(+), and, like an ordinary instrument, muted and soloed.

something like this:
- Bass drums| **m**ute **s**olo
  - bd left|  **m** **s** [select **g**roup] <- the last one is a dropdown menu
  - bd right| **m** **s** **g**

The sad thing is that we have to change the drumkit format to store group names, or use mute group numbers instead.

**Comment by theGreatWhiteShark** [comment]
Visualizing mute groups should really be done. Especially since newer kits make strong use of it. But I don't like the idea of introducing an aux channel per mute group and storing it in a drumkit file. If aux channels would ever make it into Hydrogen, the application itself could create them on fly when loading a kit. No need to store it in there.

Maybe a small icon carrying the mute group number left of the instrument's name in the sidebar of pattern editor?

But what is in an even more dire state when using mute groups is note rendering. Hydrogen does not hint at all which note will be played back when using several mute group instruments at once. Or for how long it will be played.

I did already plan to introduce a visual representation of whether a note will be played back or not in 2.0. This could be applied in here too.

Maybe also rendering note tails (the ones you get when right-click dragging a note to the right) in a dedicated color to indicate for how long a note in a mute group will be rendered.

**Comment by obitwrite** [comment]
> I did already plan to introduce a visual representation of whether a note will be played back or not in 2.0. This could be applied in here too.

This would be great. For example, when I would see a red-colored note in the pattern editor, I would be able to see which of the hihats in the hihat Mute Group is (automatically) muted. Also, for anyone not familiar with Mute Groups: when you see a red note it triggers you to check out what the color means.

**Comment by theGreatWhiteShark** [comment]
I think I would go for a more dim and grayish color compared to the black outline of a regular note. Instead of emphasizing the note that _won't_ be played back, I would move them into the background.

But this color will be adjustable like all the others in Preferences > Appearance > Color. So, you can also go for red :)

**Comment by obitwrite** [comment]
I think any color could work, so that sounds like a good idea. It should perhaps be distinguishable from the change in colors that happen when you turn down the velocity. 

The challenge I think is that when you see a note in a different color that is part of a Mute Group, that you understand what is going on. So, the relationship between the muted note and the Mute Group settings. Velocity is clearly below the note, and is something you set yourself – and you see the color change. A Mute Group is part of the drum kit settings, and is already set for you. Don't have a solution for that though, just a thought.

**Comment by theGreatWhiteShark** [comment]
> It should perhaps be distinguishable from the change in colors that happen when you turn down the velocity.

Yes. These colors are used to fill the area within the note ellipse. I intend to use a different color for the enclosing ellipse itself.

> The challenge I think is that when you see a note in a different color that is part of a Mute Group, that you understand what is going on. So, the relationship between the muted note and the Mute Group settings. Velocity is clearly below the note, and is something you set yourself – and you see the color change. A Mute Group is part of the drum kit settings, and is already set for you. Don't have a solution for that though, just a thought.

What happens during note rendering will probably we very easy to understand, as the same colors will be used when the corresponding row is muted or another one is soloed. It's basically nothing specific to the mute group itself but just hints that the note won't make a sound when reached by the playhead.

But you are right. To understand why this is happening in the first place, is not obvious. I thought of adding a small icon containing the mute group number left of the corresponding instruments in the same "muted note" color. When hovered, there could be a tool tip describe mute groups in a short sentence.

I'm in the middle of doing some big changes in the pattern editor. I will tackle this issue as soon as I'm done (maybe mid January).

**Comment by obitwrite** [comment]
> the same colors will be used when the corresponding row is muted or another one is soloed.

This is a really good idea. Looking forward to this!

> I thought of adding a small icon containing the mute group number left of the corresponding instruments in the same "muted note" color. When hovered, there could be a tool tip describe mute groups in a short sentence.

Also sounds great. Some ideas for the short sentence:
'Mute Groups mute notes that a real drummer can't play at the same time.'
'Mute Groups mimic natural drums, by muting notes that can't be played at the same time on a real kit.'
'Mute Groups mute notes that can't be played at the same time on a real drum kit.'
'Mute Groups mute notes that a drummer can't play simultaneously.'
